Tools/autotest/arduplane.py

```diff
@@ -529,6 +530,69 @@ def fly_flaps(self):
         if ex:
             raise ex
 
+    def test_throttle_failsafe(self):
+        self.context_push()
+        ex = None
+        try:
+            self.mavproxy.send('switch 6\n')
+            self.wait_mode('MANUAL')
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            receiver_bit = mavutil.mavlink.MAV_SYS_STATUS_SENSOR_RC_RECEIVER
+            self.progress("Testing receiver enabled")
+            if (not (m.onboard_control_sensors_enabled & receiver_bit)):
+                raise PreconditionFailedException()
+            self.progress("Testing receiver present")
+            if (not (m.onboard_control_sensors_present & receiver_bit)):
+                raise PreconditionFailedException()
+            self.progress("Testing receiver health")
+            if (not (m.onboard_control_sensors_health & receiver_bit)):
+                raise PreconditionFailedException()
+            self.progress("Failing receiver")
+            self.set_parameter("SIM_RC_THRFAIL", 1)
+            self.wait_mode('CIRCLE') # short failsafe
+            self.wait_mode('RTL') # long failsafe
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+            print("%s" % str(m))
+            self.progress("Testing receiver enabled")
+            if (not (m.onboard_control_sensors_enabled & receiver_bit)):
+                raise NotAchievedException()
+            self.progress("Testing receiver present")
+            if (not (m.onboard_control_sensors_present & receiver_bit)):
+                raise NotAchievedException()
+# test commented out until we fix the sensor health bit problem!
```

Comment:
What's the problem?

Reply:
Not sure now!  I can confirm that this test doesn't pass if you comment it back in, 'though.  I believe this PR was pulled out of a larger one that factored that health code up - where this bug was addressed.

Reply:
retested after commenting back in.  Must have been fixed sometime :-)

Reply:
Apparently you didn't do the exact same test because Travis failed 🙂 This is why: https://github.com/ArduPilot/ardupilot/blob/5a19f447a707e15545127769f6b8e5c2fccccb7c/ArduPlane/radio.cpp#L190
Plane is setting the time for valid RC input even when throttle is below failsafe!